test/gleam/http/request_test.gleam

```diff
@@ -178,6 +178,32 @@ pub fn get_req_header_test() {
   |> should.equal(Error(Nil))
 }
 
+pub fn get_req_case_insensitive_header_test() {
+  let make_request = fn(headers) {
+    Request(
+      method: http.Get,
+      headers: headers,
+      body: Nil,
+      scheme: http.Https,
+      host: "example.com",
+      port: None,
+      path: "/",
+      query: None,
+    )
+  }
+
+  let header_key = "GLEAM"
+  let request = make_request([#("ANSWER", "42"), #("GLEAM", "awesome")])
```

Comment:
Would be good to test the actual case insensitivity here. Right now `header_key` and the value in headers are both uppercase so it's not really testing that.

Reply:
Fixed